File: java-backend/src/main/java/com/ikunmanager/model/Announcement.java
package com.ikunmanager.model;

import lombok.Data;
import java.time.LocalDateTime;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonFormat;

@Data
public class Announcement {
    private Long id;
    private String title;
    private String content;
    @JsonProperty("author_name")
    private String authorName;
    @JsonProperty("status")
    private String status; // published / draft
    @JsonProperty("is_pinned")
    private Integer isPinned; // 0|1
    @JsonProperty("published_at")
    private LocalDateTime publishedAt;
    @JsonProperty("created_at")
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime createdAt;
    @JsonProperty("updated_at")
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime updatedAt;
} 

File: src/api/announcement.ts
import request from '@/utils/request';
import type { ApiResponse } from '@/types/common';

export interface Announcement {
  id: number;
  title: string;
  content: string;
  author_name: string;
  is_pinned: number; // 0 or 1
  published_at: string; // Formatted date string e.g., '2024-05-20 10:00'
}

/**
 * Get the list of published announcements for the student portal.
 * @param params - Optional parameters like pagination.
 * @returns A promise with the list of announcements.
 */
export function getAnnouncements(params?: any): Promise<ApiResponse<Announcement[]>> {
  return request.get<ApiResponse<Announcement[]>>('api/announcements', { params });
}

// --- Admin Functions ---

export interface AdminAnnouncement extends Announcement {
  status: 'published' | 'draft';
  updated_at: string;
}

/**
 * [Admin] Get all announcements (including drafts).
 * @returns A promise with all announcements.
 */
export function getAllAnnouncementsAdmin(): Promise<ApiResponse<AdminAnnouncement[]>> {
  return request.get<ApiResponse<AdminAnnouncement[]>>('api/announcements/all');
}

/**
 * [Admin] Create a new announcement.
 * @param data - The announcement data to create.
 * @returns A promise with the created announcement.
 */
export function createAnnouncement(data: Partial<AdminAnnouncement>): Promise<ApiResponse<Announcement>> {
  return request.post<ApiResponse<Announcement>>('api/announcements', data);
}

/**
 * [Admin] Update an existing announcement.
 * @param id - The ID of the announcement to update.
 * @param data - The new data for the announcement.
 * @returns A promise with the updated announcement.
 */
export function updateAnnouncement(id: number, data: Partial<AdminAnnouncement>): Promise<ApiResponse<Announcement>> {
  return request.put<ApiResponse<Announcement>>(`api/announcements/${id}`, data);
}

/**
 * [Admin] Delete an announcement.
 * @param id - The ID of the announcement to delete.
 * @returns A promise.
 */
export function deleteAnnouncement(id: number): Promise<ApiResponse<null>> {
  return request.delete<ApiResponse<null>>(`api/announcements/${id}`);
}